examples/vision/attention_mll_classification.py

```diff
@@ -0,0 +1,579 @@
+            img = bags[j][i]
+            fig.add_subplot(1, BAG_SIZE, j + 1)
+            plt.grid(False)
+            if attention_weights is not None:
+                plt.title(np.around(attention_weights[labels[i]][j], 2))
+            plt.imshow(img)
+        plt.show()
+
+
+# Plot some of validation data bags per class.
+plot(val_data, val_labels, "pos")
+plot(val_data, val_labels, "neg")
+
+"""
+## Create Model
+
+First we will create some embeddings per instance, invoke the attention operator and then
+use the softmax function to output the class probabilities.
+"""
+
+
+def create_model(instance_shape):
+    def dense_layers():
+
+        inputs, embeddings = [], []
+        for _ in range(BAG_SIZE):
+            inp = layers.Input(instance_shape)
+            x = layers.Flatten()(inp)
+            x = layers.Dense(126, activation="relu")(x)
+            x = layers.Dense(64, activation="relu")(x)
+            inputs.append(inp)
+            embeddings.append(x)
+
+        return inputs, embeddings
+
+    # Create layers that form embeddings.
+    inputs, embeddings = dense_layers()
+
+    # Invoke the attention layer.
+    alpha = AttentionLayer(
+        l_dim=256,
+        output_dim=1,
+        kernel_regularizer=keras.regularizers.l2(0.01),
+        use_gated=True,
+        name="alpha",
+    )(embeddings)
+
+    # Multiply attention weights with the input layers.
+    mul_layers = [layers.multiply([alpha[i], embeddings[i]]) for i in range(len(alpha))]
+
+    # Concatenate layers.
+    concat = layers.concatenate(mul_layers, axis=1)
+
+    # Classification output node.
+    output = layers.Dense(2, activation="softmax")(concat)
+
+    return keras.Model(inputs, output)
+
+
+"""
+## Class Weights
+
+Since this kind of problem could simply turn into imbalanced data classification problem,
+class weights should be considered.
+
+Let's say there are 1000 bags. There often could be cases were ~90 % of the bags does not
+contain any positive label and ~10 % does.
+Such data can be referred to as **Imbalanced data**.
+
+Using class weights, the model will tend to consider the rare class more as compared to
+that of the abundant one.
+"""
+
+
+def calculate_weights(labels):
+
+    # Count number of postive and negative bags.
+    neg_count = len(np.where(labels == 0)[0])
+    pos_count = len(np.where(labels == 1)[0])
+    total_count = neg_count + pos_count
+
+    # Build class weight dictionary.
+    return {
+        0: (1 / neg_count) * (total_count / 2),
+        1: (1 / pos_count) * (total_count / 2),
+    }
+
+
+"""
+## Build and Train Model
+
+The model is built and trained in this section.
+
+Some regularization techniques are considered to avoid overfitting the model which ensures
+minimal generalization error.
+"""
+
+
+def train(train_data, train_labels, val_data, val_labels, model):
+
+    # Train model.
+    # Prepare callbacks.
+    # Path where to save best weights.
+    with tempfile.NamedTemporaryFile(suffix=".h5") as tmp:
```

Comment:
Why not just hardcode a path?

Reply:
I don't hardcode the path because I usually prefer using "tempfile" mainly for the following reasons:

-  It creates a random file in `/tmp`, so I don't hardcode any path.
- It deletes the file once its no longer referenced.
- Its a Python built-in module. 

However, if you prefer me hardcoding the path, please let me know.

Reply:
My [latest commit](https://github.com/keras-team/keras-io/pull/593/commits/078cf800dc1193f4d22fa269f973805352929967) includes the hardcoded path. The [earlier commit](https://github.com/keras-team/keras-io/pull/593/commits/a6d8be9cc55807e00c33d672dd2c5233a4b9266f) just before it contains the same changes but with `tempfile` included.